### PR TITLE
ueberzugpp: init at 2.8.0

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

https://github.com/jstkdng/ueberzugpp/tree/master

Drop in replacement of ueberzug, the image previewer for terminal often used together with ranger-like file manager.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/ueberzugpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  cmakeFlags = lib.optionals (!withOpencv) [
+    "-DENABLE_OPENCV=OFF"
+  ] ++ lib.optionals (!withX11) [
+    "-DENABLE_X11=OFF"
+  ];
+
+  meta = with lib; {
+    description = "Drop in replacement for ueberzug written in C++";
+    homepage = "https://github.com/jstkdng/ueberzugpp";
+    license = licenses.gpl3Only;
```
Comment (figsoda) [inline pkgs/tools/graphics/ueberzugpp/default.nix:68]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/tools/graphics/ueberzugpp/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  ] ++ lib.optionals (!withX11) [
+    "-DENABLE_X11=OFF"
+  ];
+
+  meta = with lib; {
+    description = "Drop in replacement for ueberzug written in C++";
+    homepage = "https://github.com/jstkdng/ueberzugpp";
+    license = licenses.gpl3Plus;
+    mainProgram = "ueberzug";
+    maintainers = with maintainers; [ aleksana ];
```
Comment (wegank) [inline pkgs/tools/graphics/ueberzugpp/default.nix:70]:
```suggestion
    maintainers = with maintainers; [ aleksana ];
    broken = stdenv.isDarwin && stdenv.isx86_64;
```

Comment (Aleksanaa) [reply pkgs/tools/graphics/ueberzugpp/default.nix:70]: Did you test aarch64 Darwin? I think `platforms.linux` may be okay?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 230692` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/tools/graphics/ueberzugpp/default.nix (hunk 2 above)
Comment (wegank) [inline pkgs/tools/graphics/ueberzugpp/default.nix:70]:
```suggestion
    maintainers = with maintainers; [ aleksana ];
    platforms = platforms.unix;
```

Forgot this.

**Review by figsoda — APPROVED** [review]

**Pull request merged**